cpp/CMakeLists.txt

```diff
@@ -637,19 +602,31 @@ if(compiled IN_LIST raft_FIND_COMPONENTS)
 endif()
 ]=]
 )
+set(raft_components compiled distributed)
+set(raft_export_sets raft-compiled-exports raft-distributed-exports)
+if(TARGET raft_lib)
+  list(APPEND raft_components compiled)
+  list(APPEND raft_export_sets raft-compiled-lib-exports)
```

Comment:
At this point I'm a bit confused by the different targets and export sets that we have in raft. Why do we need both the `raft_compiled` and `raft_lib` targets? It used to be necessary for the nn/dist targets so that calling code could use either component independently as header-only or precompiled, but now why can't the `raft` target just be the header-only target and `raft_compiled` always be precompiled? And that stems from my other question, why do we need both `raft-compiled-exports` and `raft-compiled-lib-exports`?

Reply:
@vyasr from my understanding at this point, the answer is "we don't need both `raft_compiled` and `raft_lib` anymore, but at the current time they aren't really causing harm, so we aren't in an immediate rush to remove them". @robertmaynard please correct me if that assertion is wrong.

Reply:
The `raft_compiled` target is created even when `RAFT_COMPILE_LIBRARY` is turned off, so one solution for proper invariant behavior when generating the libraft-headers and libraft was to have the current organization.

If we are okay with making the `raft::compiled` target conditional all this logic can be removed and we could have a single target `raft_compiled` and export set ( `raft-compiled-exports` ). If we are good with this, I would be happy to open a PR that does these changes.